### ISSUE TITLE
Update Github workflow

### ISSUE THREAD

**Issue by aferrar** [NYCPlanning/edm-overview#734]
Met with Data Engineering to discuss various ways of improving github workflow. Following suggestions should be implemented for all current GIS scripts:

- [ ] Use relative paths for calling configuration files (probably stick with .ini for convenience) within scripts
- [ ] Use gitignore to omit unneeded files/folders from repo (real ini, documentation, logs, etc)
- [ ] Use private repos while establishing workflow so as to not accidentally expose internal paths. Can push to a public repo later if deemed appropriate/useful/necessary
- [ ] Use symbolic links to maintain code between network location and local drive where code will be pushed from
- [ ] Optional: Add template ini files

**Comment by aferrar** [comment]
Attempted to create symlinks between network path (using UNC path) and local drive but received following error:

Local volumes are required to complete the operation.

Believe I do not have adequate permission to create network symbolic links. Need to reach out to PCNS to have this done for me...

**Comment by aferrar** [comment]
Reached out to PCNS on 4/5 regarding symbolic links. Waiting for response.

**Comment by aferrar** [comment]
Followed up on request to PCNS for symlinks. Waiting for response.

**Comment by aferrar** [comment]
Need to refactor following projects in prep for republishing to associated repos and continued maintenance

**Non-Scheduled**
- [ ] DCM
- [ ] DigitalOcean
- [ ] E-Designations
- [ ] Housing_Dev_DB_Distribute
- [ ] LION_Disttribute
- [ ] MapPLUTO_Distribute
- [ ] MIH_Distribute
- [ ] Notifications
- [ ] POPS
- [ ] WPAA

**Scheduled**
- [ ] E-Designations
- [ ] LDGMA
- [ ] Building Footprints
- [ ] CSCL
- [ ] LPC
- [ ] SDE Maintenance

**Comment by aferrar** [comment]
E-Designations:
- Refactored multiple directories into a single consolidated directory for easier commiting
- Created gitignore for files/folders/IDE configs that will remain local
- Changed all absolute paths to relative where applicable
- Split distribution script into two (BP and SDE) per {insert issue with all code modifications here when created}
    - Still need to make additional modifications. See above issue link
- Added VSCode workspace config in addition to already existing PyCharm config

**Comment by caseysmithpgh** [comment]
Questions for DE: 

1. How does having our local repos on a share drive impact branch pull downs?

**Comment by damonmcc** [comment]
hey @caseysmithpgh! some impacts that come to my mind are:
- when trying to pull down updates to a local repo stored on a shared drive (effectively edit some files), there may be an issue when multiple people try it at the same time
- if you're running scripts that can change because someone else ran a git command (`commit`, `pull`, `revert`) that could be disruptive/confusing

in some sense, the remote repo on github _is_ a shared drive. different people have local copies of the repo that only they can edit (including the different branches they may want to see locally). and when new commits are pushed/pulled to/from remote, conflicts are detected and can be resolved

ideally, people are working on different branches or proactive about not making conflicting commits to the same branch

hope I'm understanding the question and those thoughts made sense! happy to chat whenever

**Comment by caseysmithpgh** [comment]
@damonmcc Thanks for the feedback! It sounds like best practice for dev branches would be to pull into a directory that does not live on a shared drive, and then following a merge we would need to pull updates to the shared drive. Does that sound like a logical approach?

**Comment by damonmcc** [comment]
@caseysmithpgh yup that works! but I'd add that you could pull dev branch updates down to the shared drive before any branches (`dev-jack-1` to `main`, `dev-1` to `dev-2`, etc.) have been merged together. conceptually, I'm imagining the shared drive as a distinct person who has it's own copy of the repo

also I think I'm missing why the code should or must also live on the shared drive. maybe something to do with ESRI software access to it?

**Comment by croswell81** [comment]
Expired
